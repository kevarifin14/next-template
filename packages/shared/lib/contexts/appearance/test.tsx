import { render, fireEvent } from '@testing-library/react';
import React from 'react';

import { AppearanceProvider, useAppearanceContext } from '.';

describe('Apprearance Context', () => {
  const AppearanceConsumer = () => {
    const { dark, setDark } = useAppearanceContext();
    return (
      <>
        <button onClick={() => setDark((prev) => !prev)}>Toggle Dark Mode</button>
        {dark ? 'dark' : 'light'}
      </>
    );
  };

  const wrapper = ({ children }) => <AppearanceProvider dark>{children}</AppearanceProvider>;

  it('matches snapshot', () => {
    const { baseElement } = render(<AppearanceConsumer />, { wrapper });
    expect(baseElement).toMatchSnapshot();
  });

  it('toggles dark mode', () => {
    const { getByRole, queryByText } = render(<AppearanceConsumer />, { wrapper });

    const toggleDarkModeButton = getByRole('button');

    expect(queryByText('dark')).toBeTruthy();

    fireEvent.click(toggleDarkModeButton);

    expect(queryByText('light')).toBeTruthy();
  });
});
