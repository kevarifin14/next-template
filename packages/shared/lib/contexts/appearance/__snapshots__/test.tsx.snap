// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apprearance Context matches snapshot 1`] = `
<body>
  <div>
    <div
      class="dark"
    >
      <div
        class="transform transition-colors h-full bg-light dark:bg-dark"
      >
        <button>
          Toggle Dark Mode
        </button>
        dark
      </div>
    </div>
  </div>
</body>
`;
